src/bm_sim/actions.cpp

```diff
@@ -289,6 +290,8 @@ ActionFnEntry::execute(Packet *pkt) const {
   auto &primitives = action_fn->primitives;
   size_t param_offset = 0;
   // primitives is a vector of pointers
+  const SourceInfo *si = action_fn->get_source_info();
```

Comment:
put it in the macro below so there is no penalty when logging macros are disabled